### PR TITLE
Dm create comments

### PR DESCRIPTION
**Pull request by DMJr**
@HeroicEric  KILLING IT KID.. ALL DAY

### CODE REVIEW

**Comment by townie** [comment]
@HeroicEric Can we get a code review?

File: app/controllers/mugshots_controller.rb [hunk 1]
```diff
@@ -15,6 +16,8 @@ def create
 
   def show
     @mugshot = Mugshot.find(params[:id])
+    @mugshot_comments = Comment.where("mugshot_id = ?", @mugshot)
```
Comment (HeroicEric) [inline app/controllers/mugshots_controller.rb]: You should use the activerecord syntax when it's possible

``` ruby
@mugshot.comments
```

File: app/controllers/mugshots_controller.rb [hunk 2]
```diff
@@ -22,11 +25,16 @@ def index
   end
 
   def edit
-    @mugshot = Mugshot.find(params[:id])
+    if current_user.mugshots.exists?(params[:id])
```
Comment (HeroicEric) [inline app/controllers/mugshots_controller.rb:28]: You don't need to check this. Rails will just send them a 404 if the record is not found.

File: spec/features/delete_a_mugshot_spec.rb [hunk 3]
```diff
@@ -13,17 +13,24 @@
...
-    click_button 'Create Mugshot'
 
-    expect(page).to have_content('New Mugshot Submitted!')
-    expect(page).to have_content('Bieber')
+    celeb = FactoryGirl.create(:mugshot, user: user)
+    celeb.save
 
-    click_link 'Delete Mugshot'
+    # fill_in 'First Name', with: celeb.first_name
+    # fill_in 'Last Name', with: celeb.last_name
```
Comment (HeroicEric) [inline spec/features/delete_a_mugshot_spec.rb]: Don't commit commented out code

File: app/views/comments/new.html.erb [hunk 4]
```diff
@@ -0,0 +1 @@
+<h1>Comment bad</h1>
```
Comment (HeroicEric) [inline app/views/comments/new.html.erb]: ?

File: app/models/mugshot.rb [hunk 5]
```diff
@@ -1,4 +1,7 @@
 class Mugshot < ActiveRecord::Base
+  belongs_to :user
+  has_many :comments
```
Comment (HeroicEric) [inline app/models/mugshot.rb]: What kind of dependencies should this have?

File: app/models/user.rb [hunk 6]
```diff
@@ -4,6 +4,9 @@ class User < ActiveRecord::Base
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
 
+  has_many :mugshots
```
Comment (HeroicEric) [inline app/models/user.rb]: What kind of dependencies should this have?

File: app/models/user.rb [hunk 7]
```diff
@@ -4,6 +4,9 @@ class User < ActiveRecord::Base
...
+  has_many :comments
```
Comment (HeroicEric) [inline app/models/user.rb]: What kind of dependencies should this have?

File: app/views/layouts/application.html.erb [hunk 8]
```diff
@@ -24,6 +24,7 @@
       <section class="top-bar-section">
         <ul class="right">
           <% if user_signed_in? %>
+            <li > <%= link_to "Mugshots", mugshots_path %></li>
```
Comment (HeroicEric) [inline app/views/layouts/application.html.erb]: Do we want to force people to sign up just to view the mugshots?

File: app/views/mugshots/index.html.erb [hunk 9]
```diff
@@ -1,8 +1,27 @@
 <h1> Welcome to the Mugshot Index Page </h1>
 
-<% @mugshots.each do |mugshot| %>
+<a href=<%= new_mugshot_path %> class="button expand">Create a MUGSHOT</a>
```
Comment (HeroicEric) [inline app/views/mugshots/index.html.erb]: Use `link_to`

File: app/views/mugshots/index.html.erb [hunk 10]
```diff
@@ -1,8 +1,27 @@
...
+  <fieldset>
+    <div class="row">
+      <div class="row">
+        <div class="small-12 medium-12 columns">
+          <%= link_to mugshot.full_name, mugshot_path(mugshot) %>
+        </div>
+      </div>
+      <div class="row">
+        <div class="small-12 medium-12 columns">
+          <%= "PhotoHere" %>
```
Comment (HeroicEric) [inline app/views/mugshots/index.html.erb]: ?

File: app/views/mugshots/index.html.erb [hunk 11]
```diff
@@ -1,8 +1,27 @@
...
 
-<% @mugshots.each do |mugshot| %>
+<a href=<%= new_mugshot_path %> class="button expand">Create a MUGSHOT</a>
+
 
-  <p><%= mugshot.first_name %></p>
-  <p><%= mugshot.last_name %></p>
+<% @mugshots.each do |mugshot| %>
 
+  <fieldset>
```
Comment (HeroicEric) [inline app/views/mugshots/index.html.erb]: `<fieldset>` tags should only be used inside of `<form>` tags.

File: app/views/mugshots/index.html.erb [hunk 12]
```diff
@@ -1,8 +1,27 @@
...
+        </div>
+      </div>
+      <div class="row">
+        <div class="small-12 medium-12 columns">
+          <%= "PhotoHere" %>
+        </div>
+      </div>
+      <div class="row">
+        <div class="small-12 medium-12 columns">
+          <%= "mugshot.comments" %>
```
Comment (HeroicEric) [inline app/views/mugshots/index.html.erb:20]: ?

File: db/migrate/20140408162355_add_userid_to_mugshots.rb [hunk 13]
```diff
@@ -0,0 +1,8 @@
+class AddUseridToMugshots < ActiveRecord::Migration
+  def change
+    add_column :mugshots, :user_id, :integer, null: false
+    add_index :mugshots, :user_id
+    add_index :comments, :mugshot_id
```
Comment (HeroicEric) [inline db/migrate/20140408162355_add_userid_to_mugshots.rb:5]: Nice!

File: spec/features/edit_an_existing_mugshot_spec.rb [hunk 14]
```diff
@@ -8,19 +8,45 @@
...
   scenario 'edit an existing mugshot' do
-    mugshot = FactoryGirl.create(:mugshot)
-    visit edit_mugshot_path(mugshot)
+    user = FactoryGirl.create(:user)
+    sign_in_as(user)
+    visit new_mugshot_path
+
+    celeb = FactoryGirl.create(:mugshot, user: user)
+    celeb.save
+
```
Comment (HeroicEric) [inline spec/features/edit_an_existing_mugshot_spec.rb]: Extra lines

File: spec/features/edit_an_existing_mugshot_spec.rb [hunk 15]
```diff
@@ -8,19 +8,45 @@
...
-    fill_in 'Last Name', with: 'Davidson'
+    fill_in 'Last Name', with: "Davidson"
 
-    click_button 'Update Mugshot'
+    click_button 'Update'
     expect(page).to have_content('Davidson')
     expect(page).to have_content('Mugshot Updated')
   end
+
+  scenario "An invalid user can not edit a mughshot that doesnot belong to them" do
```
Comment (HeroicEric) [inline spec/features/edit_an_existing_mugshot_spec.rb]: `scenario "user tries to edit another user's mugshot"`

The user isn't invalid.

File: spec/models/comment_spec.rb [hunk 16]
```diff
@@ -0,0 +1,13 @@
+require 'spec_helper'
+
+describe Comment do
+
```
Comment (HeroicEric) [inline spec/models/comment_spec.rb:4]: Remove this line or add one at the bottom of the method. Either way, the idea is to be consistent.

**Comment by HeroicEric** [comment]
@LaunchAcademy/celebrity-mugshot-reviews left some comments. Let me know when they have been addressed.

File: app/controllers/mugshots_controller.rb (hunk 2 above)
Comment (DMJr) [inline app/controllers/mugshots_controller.rb:28]: @HeroicEric well dont we want for better UX we redirect them to index?

File: app/views/comments/new.html.erb (hunk 4 above)
Comment (DMJr) [inline app/views/comments/new.html.erb]: just had to redirect to a page

File: app/controllers/mugshots_controller.rb (hunk 2 above)
Comment (HeroicEric) [inline app/controllers/mugshots_controller.rb:28]: Maybe. I think the best UX would be to not give them a way to end up on a page that they don't have access to. I guess you can leave it if you really want to.

File: app/views/mugshots/index.html.erb (hunk 9 above)
Comment (DMJr) [inline app/views/mugshots/index.html.erb]: we did it for styling purposeds BIG BUTTON! GET EXCITED

File: app/views/mugshots/index.html.erb (hunk 10 above)
Comment (DMJr) [inline app/views/mugshots/index.html.erb]: display a mugshot and all the comments assocated with tat mugshot and you can create a new comment all on the same pAGE GET EXCITED!~!!~!!~!@#@!#!@

**Comment by DMJr** [comment]
@HeroicEric DO IUT

File: app/views/mugshots/index.html.erb (hunk 9 above)
Comment (HeroicEric) [inline app/views/mugshots/index.html.erb]:
``` ruby
<%= link_to "Create a Mugshot", new_mugshot_path, class: "button expand" %>
```

File: spec/models/comment_spec.rb (hunk 16 above)
Comment (DMJr) [inline spec/models/comment_spec.rb:4]: I DONT GIVE A DIFF

**Comment by DMJr** [comment]
@HeroicEric does this meet ur minimum?

File: spec/features/delete_a_mugshot_spec.rb [hunk 17]
```diff
@@ -13,17 +13,16 @@
...
-    fill_in 'First Name', with: 'Justin'
-    fill_in 'Last Name', with: 'Bieber'
-    fill_in 'Description', with: 'This guy should not be drunk driving'
-    select "2014" && "April" && '11', from: "Approximate Date"
-    click_button 'Create Mugshot'
 
-    expect(page).to have_content('New Mugshot Submitted!')
-    expect(page).to have_content('Bieber')
+    celeb = FactoryGirl.create(:mugshot, user: user)
+    celeb.save
```
Comment (HeroicEric) [inline spec/features/delete_a_mugshot_spec.rb]: Do we need to save this twice?

File: spec/features/edit_an_existing_mugshot_spec.rb [hunk 18]
```diff
@@ -5,22 +5,45 @@
...
 
   scenario 'edit an existing mugshot' do
-    mugshot = FactoryGirl.create(:mugshot)
-    visit edit_mugshot_path(mugshot)
+    user = FactoryGirl.create(:user)
+    sign_in_as(user)
+    visit new_mugshot_path
+
+    celeb = FactoryGirl.create(:mugshot, user: user)
+    celeb.save
```
Comment (HeroicEric) [inline spec/features/edit_an_existing_mugshot_spec.rb]: Do we need to save this twice?

File: spec/models/mugshot_spec.rb [hunk 19]
```diff
@@ -19,5 +19,7 @@
     it 'validates imaging format'
     it 'validates url formatting'
 
+    it {should have_many(:comments)}
```
Comment (HeroicEric) [inline spec/models/mugshot_spec.rb:22]:
``` ruby
it { should have_many(:comments).dependent(:destroy) }
```

File: spec/models/mugshot_spec.rb [hunk 20]
```diff
@@ -19,5 +19,7 @@
...
+    it {should belong_to(:user)}
```
Comment (HeroicEric) [inline spec/models/mugshot_spec.rb:23]: Same ^^

File: spec/models/user_spec.rb [hunk 21]
```diff
@@ -64,4 +64,6 @@
     expect(valid_user.full_name).to eq("Gene Simons")
   end
 
+  it {should have_many(:comments)}
```
Comment (HeroicEric) [inline spec/models/user_spec.rb:67]: You can also specify dependencies here

**Comment by DMJr** [comment]
@HeroicEric  how's it look

**Pull request merged**